src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,297 @@
+package duke.util;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+
+/**
+ * Parses Duke's commands.
+ *
+ * <p>Interprets and converts raw input from the command line to instances of
+ * {@link Command} to be executed by Duke.</p>
+ */
+public class Parser {
+    private static final List<String> DATE_FORMAT_STRINGS = Arrays.asList(
+        "dd/MM/yyyy HH:mm", "dd-MM-yyyy HH:mm", "yyyy/MM/dd HH:mm", "yyyy-MM-dd HH:mm",
+        "dd/MM/yyyy", "dd-MM-yyyy", "yyyy/MM/dd", "yyyy-MM-dd",
+        "dd/MM HH:mm", "dd-MM HH:mm", "dd/MM", "dd-MM");
+
+    /**
+     * Enumerates all task types known to the parser.
+     */
+    private enum TaskType {
+        TODO("todo", null),
+        DEADLINE("deadline", "/by"),
+        EVENT("event", "/at");
+
+        public final String commandPhrase;
+        public final String timeDivider;
+
+        TaskType(String commandPhrase, String timeDivider) {
+            this.commandPhrase = commandPhrase;
+            this.timeDivider = timeDivider;
+        }
+
+        /**
+         * Gets the corresponding TaskType using its command phrase.
+         */
+        static TaskType getTaskType(String commandPhrase) {
+            Optional<TaskType> result = Arrays.stream(TaskType.values())
+                    .filter(t -> t.commandPhrase.equals(commandPhrase))
+                    .findFirst();
+
+            if (result.isEmpty()) {
+                throw new RuntimeException("Our parser encountered a fatal error.");
+            }
+
+            return result.get();
+        }
+    }
+
+    private boolean isBeginWithVowel(String string) {
```

Comment:
Nice attention to detail in using this to ensure grammar is correct 👍🏻 . This idea could also be used for singular and plural words as well.